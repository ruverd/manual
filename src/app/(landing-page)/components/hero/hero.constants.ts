import heroBg from '@/assets/hero-bg.png';
import logo from '@/assets/manual-logo.svg';

export const COMPONENT_MAP = {
  title: {
    lines: ['Be good', 'to yourself']
  },
  description:
    "We're working around the clock to bring you a holistic approach to your wellness. From top to bottom, inside and out.",
  button: {
    text: 'Take the Quiz',
    href: '/quiz'
  },
  background: {
    src: heroBg,
    alt: 'Man smiling'
  },
  logo: {
    src: logo,
    alt: 'Manual Logo'
  }
};
