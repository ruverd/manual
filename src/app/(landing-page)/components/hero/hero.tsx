import Image from 'next/image';
import Link from 'next/link';

import { Button } from '@/components/ui';
import { COMPONENT_MAP } from './hero.constants';

export const Hero = () => {
  return (
    <div className="relative">
      <Image
        src={COMPONENT_MAP.background.src}
        alt={COMPONENT_MAP.background.alt}
        className="size-full object-cover"
        placeholder="blur"
        priority
      />
      <Image
        src={COMPONENT_MAP.logo.src}
        alt={COMPONENT_MAP.logo.alt}
        className="absolute left-6 top-4 size-6 sm:left-12 md:left-16 md:top-[30px] md:size-10 lg:left-[138px]"
      />
      <div className="absolute inset-0 mx-6 flex flex-col sm:mx-12 md:mx-16 lg:mx-[138px]">
        <div className="my-auto flex w-full max-w-52 flex-col gap-2 pt-4 sm:max-w-72 md:max-w-96 md:gap-5 md:pt-12">
          <h1 className="text-4xl leading-none tracking-neg-3 text-brand md:text-5xl lg:text-[90px]">
            <span className="block whitespace-nowrap">
              {COMPONENT_MAP.title.lines[0]}
            </span>
            <span className="block whitespace-nowrap">
              {COMPONENT_MAP.title.lines[1]}
            </span>
          </h1>
          <p className="hidden text-balance text-sm leading-6 sm:block md:text-lg md:leading-[30px]">
            {COMPONENT_MAP.description}
          </p>
          <Button className="mt-4 w-fit" asChild>
            <Link
              href={COMPONENT_MAP.button.href}
              aria-label={COMPONENT_MAP.button.text}
              tabIndex={0}
            >
              {COMPONENT_MAP.button.text}
            </Link>
          </Button>
        </div>
      </div>
    </div>
  );
};
