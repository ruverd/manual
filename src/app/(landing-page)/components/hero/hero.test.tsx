import { render, screen } from '@testing-library/react';
import { beforeEach, describe, expect, it, vi } from 'vitest';

import { Hero } from '.';
import { COMPONENT_MAP } from './hero.constants';

vi.mock('next/image', () => ({
  default: vi.fn(({ alt, className }) => (
    <img alt={alt} className={className} data-testid="mock-image" />
  ))
}));

const renderComponent = () => {
  return render(<Hero />);
};

describe('Hero', () => {
  beforeEach(() => {
    renderComponent();
  });

  it('should render the title with correct text', () => {
    COMPONENT_MAP.title.lines.forEach(line => {
      expect(screen.getByText(line)).toBeInTheDocument();
    });
  });

  it('should render the description text', () => {
    const description = screen.getByText(COMPONENT_MAP.description);

    expect(description).toBeInTheDocument();
  });

  it('should render the background image', () => {
    expect(
      screen.getByAltText(COMPONENT_MAP.background.alt)
    ).toBeInTheDocument();
  });

  it('should render the logo image', () => {
    expect(screen.getByAltText(COMPONENT_MAP.logo.alt)).toBeInTheDocument();
  });

  it('should render the CTA button with correct link', () => {
    const buttonLink = screen.getByRole('link', {
      name: COMPONENT_MAP.button.text
    });

    expect(buttonLink).toHaveTextContent(COMPONENT_MAP.button.text);
    expect(buttonLink).toHaveAttribute('href', COMPONENT_MAP.button.href);
  });

  it('should match snapshots', () => {
    const { container } = render(<Hero />);

    expect(container).toMatchSnapshot();
  });
});
