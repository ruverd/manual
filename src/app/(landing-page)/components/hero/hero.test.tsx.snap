// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Hero > should match snapshots 1`] = `
<div>
  <div
    class="relative"
  >
    <img
      alt="Man smiling"
      class="size-full object-cover"
      data-testid="mock-image"
    />
    <img
      alt="Manual Logo"
      class="absolute left-6 top-4 size-6 sm:left-12 md:left-16 md:top-[30px] md:size-10 lg:left-[138px]"
      data-testid="mock-image"
    />
    <div
      class="absolute inset-0 mx-6 flex flex-col sm:mx-12 md:mx-16 lg:mx-[138px]"
    >
      <div
        class="my-auto flex w-full max-w-52 flex-col gap-2 pt-4 sm:max-w-72 md:max-w-96 md:gap-5 md:pt-12"
      >
        <h1
          class="text-4xl leading-none tracking-neg-3 text-brand md:text-5xl lg:text-[90px]"
        >
          <span
            class="block whitespace-nowrap"
          >
            Be good
          </span>
          <span
            class="block whitespace-nowrap"
          >
            to yourself
          </span>
        </h1>
        <p
          class="hidden text-balance text-sm leading-6 sm:block md:text-lg md:leading-[30px]"
        >
          We're working around the clock to bring you a holistic approach to your wellness. From top to bottom, inside and out.
        </p>
        <a
          aria-label="Take the Quiz"
          class="inline-flex h-8 items-center justify-center whitespace-nowrap bg-primary px-[30px] py-[15px] text-[10px] font-bold uppercase tracking-[0.15em] text-primary-foreground transition-colors hover:bg-primary/90 focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-20 md:h-[45px] [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 mt-4 w-fit"
          href="/quiz"
          tabindex="0"
        >
          Take the Quiz
        </a>
      </div>
    </div>
  </div>
</div>
`;
